apps/meteor/client/views/teams/contextualBar/channels/RoomActions.js

```diff
@@ -19,16 +19,18 @@ const useReactModal = (Component, props) => {
 	});
 };
 
-const RoomActions = ({ room, reload }) => {
+const RoomActions = ({ room, mainRoomId, reload }) => {
 	const t = useTranslation();
 	const rid = room._id;
 	const type = room.t;
 
 	const dispatchToastMessage = useToastMessageDispatch();
 
-	const canDeleteTeamChannel = usePermission(type === 'c' ? 'delete-c' : 'delete-p', rid);
-	const canEditTeamChannel = usePermission('edit-team-channel', rid);
-	const canRemoveTeamChannel = usePermission('remove-team-channel', rid);
+	const canDeleteChannel = usePermission(`delete-${type}`, rid);
+	const canDeleteTeamChannel = usePermission(`delete-team-${type === 'c' ? 'channel' : 'group'}`, mainRoomId);
+	const canDelete = canDeleteChannel && canDeleteTeamChannel;
```

Comment:
Same as before, do we need both permissions combined to be able to delete the channel?

Reply:
In this case it's safe to assume we need it since this component is only used for rooms inside teams.